artcommon/artcommonlib/konflux/konflux_db.py

```diff
@@ -170,7 +170,7 @@ async def get_latest_builds(self, names: typing.List[str], group: str,
 
     def get_latest_build(self, name: str, group: str, outcome: KonfluxBuildOutcome = KonfluxBuildOutcome.SUCCESS,
                          assembly: str = 'stream', el_target: str = None, artifact_type: ArtifactType = None,
-                         engine: Engine = None, completed_before: datetime = None)\
+                         engine: Engine = None, completed_before: datetime = None, extra_patterns: dict = {})\
```

Comment:
While this works, I think it would be nice to add commit as a first class search param (something like `git_short_commit`). Then the fact that the short commit is part of the release field can remain the funcs internal implementation which the caller doesn't need to know.

Reply:
Also this could be different for bundle builds since I think `version` is where the git commit resides..

Reply:
Or we just implement generic support for nvr pattern, then the caller doesn't need to know which field, but just the intent of "search nvrs with this in nvr name"

Reply:
`extra_patterns` sounds too generic to me either. Maybe we need to rename it or make those fields for search as first-class parameters. otherwise it looks good to me.

Reply:
In the unit tests I have probably provided a misleading example. Commitish is being stored in its on field in the DB, and it's also part of the NVR (that we're already storing). 

I deliberately made this generic to give us extra flexibility, so `extra_patterns` was named after this idea. If you have a better name I can replace it, but I would still keep the generic thing. Then, I can give the `commitish` search a greater dignity, by making it explicit like Sid suggested. We'll be searching in the dedicated field anyway, since the DB was designed to state the upstream commitish explicitly for successful builds.

Reply:
@thegreyd we're already storing the NVR. To do stuff like "nvr contains this commit" we still need this PR in. Keeping it generic will give us the greatest flexibility. I renamed the param to `search_patterns` to make it clearer. Merging as-is, then I will open a follow up PR to make the commitish search a first class citizen as you suggested